test/rbacHierarchicalDomain.test.ts

```diff
@@ -0,0 +1,98 @@
+import { DefaultRoleManager, newModel, newSyncedEnforcer, RoleManager, SyncedEnforcer } from '../src';
+
+describe('Test Hierarchical Domain RBAC', () => {
+  let e: SyncedEnforcer;
+  beforeAll(async () => {
+    const m = newModel();
+    m.addDef('r', 'r', 'sub, dom, obj, act');
+    m.addDef('p', 'p', 'sub, dom, obj, act');
+    m.addDef('g', 'g', '_, _, _');
+    m.addDef('g', 'g2', '_, _');
+    m.addDef('e', 'e', 'some(where (p.eft == allow))');
+    m.addDef('m', 'm', 'g(r.sub, p.sub, r.dom) && g2(r.dom, p.dom) && r.obj == p.obj && regexMatch(r.act, p.act)');
+
+    e = await newSyncedEnforcer(m);
+
+    await e.addPermissionForUser('admin', 'domain1', 'data1', '(read|write)');
+    await e.addPermissionForUser('admin', 'domain2', 'data2', '(read|write)');
+    await e.addPermissionForUser('admin', 'domain3', 'data3', '(write)');
+    await e.addPermissionForUser('admin', 'sibling2', 'sdata2', '(read|write)');
+
+    await e.addNamedGroupingPolicy('g', 'alice', 'admin', 'domain1');
+    await e.addNamedGroupingPolicy('g', 'bob', 'admin', 'domain2');
+    await e.addNamedGroupingPolicy('g2', 'domain1', 'domain2');
```

Comment:
These are defined in 'g2, parent, child' order. Idea here is that `parent` inherits all the `child`'s policies. Without domain hierarchy user would have to know the original domain where the policy was defined. When hierarchy is added, user can just call whichever domain in the hierarchy.

Reply:
You can add this in the docs :+1: 

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :ok_hand: This code looks good&nbsp; | &nbsp;**Tags:** Reusable